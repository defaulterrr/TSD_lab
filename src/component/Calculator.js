import React from 'react'
import ValueBox from './ValueBox/ValueBox'
import ActionButton from './ActionButton/ActionButton'
import calculatorModel from './../model/CalculatorModel'

class Calculator extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            firstValue:     0,
            secondValue:    0,
            result:         0,
        }
        this.onPress = this.onPress.bind(this)
        
    }

    onPress = (label) => {
        const num1 = Number(document.getElementById("num_1").value)
        const num2 = Number(document.getElementById("num_2").value)
        this.setState({
            result:     calculatorModel(label,num1,num2)
        }) 
    }  
    
    render() {
        return (
            <div className="Calculator">
                <ValueBox label=" First Value: " id="num_1" value={this.state.firstValue}></ValueBox>
                <ValueBox label="Second Value: " id="num_2" value={this.state.secondValue}></ValueBox>
                <ActionButton label="+" action={this.onPress}></ActionButton>
                <ActionButton label="-" action={this.onPress}></ActionButton>
                <ActionButton label="/" action={this.onPress}></ActionButton>
                <ActionButton label="*" action={this.onPress}></ActionButton>
        <h1>Answer: {this.state.result}</h1>
            </div>
        )
    }
}

export default Calculator;