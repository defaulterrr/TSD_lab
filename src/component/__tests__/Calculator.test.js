import React, { createElement } from 'react'
import { create } from 'react-test-renderer'

import Calculator from './../Calculator'

// Still makes sense to see if this Component actually renderable
describe("[RENDER] Calculator – Snapshot test", ()=> {
    test("Matches the snapshot", ()=>{
        const calculator = create(<Calculator/>)
        expect(calculator.toJSON()).toMatchSnapshot()
    })
})

// Now to some end-to-end testing

describe("[FUNCTIONAL] Calculator – a+b",()=>{
    test("It shows the expected result",()=>{
        const component = create(<Calculator/>)
        const instance = component.root
        const buttons = instance.findAllByType("button")
        const plusButton = buttons[0]
        component.props.setState({
               firstValue:  2,
               secondValue: 3    
            }
        )
        plusButton.props.onClick()
    })
})