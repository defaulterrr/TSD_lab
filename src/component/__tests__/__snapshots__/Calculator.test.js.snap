// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Calculator â€“ Snapshot test Matches the snapshot 1`] = `
<div
  className="Calculator"
>
  <div
    class="ValueBox"
    className="num_1"
  >
    <label
      className="inputLabel"
    >
       First Value: 
    </label>
    <input
      className="inputValue"
      id="num_1"
      onChange={[Function]}
      type="text"
      value={0}
    />
  </div>
  <div
    class="ValueBox"
    className="num_2"
  >
    <label
      className="inputLabel"
    >
      Second Value: 
    </label>
    <input
      className="inputValue"
      id="num_2"
      onChange={[Function]}
      type="text"
      value={0}
    />
  </div>
  <div
    class="ActionButton"
    className="ActionButton"
  >
    <button
      class="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    class="ActionButton"
    className="ActionButton"
  >
    <button
      class="button"
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <div
    class="ActionButton"
    className="ActionButton"
  >
    <button
      class="button"
      onClick={[Function]}
    >
      /
    </button>
  </div>
  <div
    class="ActionButton"
    className="ActionButton"
  >
    <button
      class="button"
      onClick={[Function]}
    >
      *
    </button>
  </div>
  <h1>
    Answer: 
    0
  </h1>
</div>
`;
